warp10/src/main/java/io/warp10/standalone/StandaloneScriptRunner.java

```diff
@@ -224,19 +238,44 @@ public void run() {
             m.appendTail(mc2WithReplacement);
 
             stack.execMulti(mc2WithReplacement.toString());
+
+            // Did the user asked to reschedule script to another period with RUNNERNEXT ?
+            if (stack.getAttribute(WarpScriptStack.ATTRIBUTE_RUNNER_RESCHEDULE_PERIOD) != null) {
+              periodicityForNextRun = (Long) stack.getAttribute(WarpScriptStack.ATTRIBUTE_RUNNER_RESCHEDULE_PERIOD);
+            }
+            // Did the user asked to reschedule script to another absolute time with RUNNERPROG ? (absolute milliseconds)
+            if (stack.getAttribute(WarpScriptStack.ATTRIBUTE_RUNNER_RESCHEDULE_TIMESTAMP) != null) {
+              progForNextRun = (Long) stack.getAttribute(WarpScriptStack.ATTRIBUTE_RUNNER_RESCHEDULE_TIMESTAMP);
+            }
           } catch (Exception e) {
             Sensision.update(SensisionConstants.SENSISION_CLASS_WARPSCRIPT_RUN_FAILURES, labels, 1);
           } finally {
             WarpConfig.clearThreadProperties();
             WarpScriptStackRegistry.unregister(stack);
             currentThread().setName(name);
-            nextrun.put(script, nowns + periodicity * 1000000L);
+            // manage possible overflow (MAXLONG RUNNERNEXT)
+            long progByPeriod = Long.MAX_VALUE;
+            if (periodicityForNextRun < (Long.MAX_VALUE / 1000000L) && (nowns + periodicityForNextRun * 1000000L) > nowns) {
+              progByPeriod = nowns + periodicityForNextRun * 1000000L;
+            }
+
+            // progForNextRun is absolute time, not jvm relative time
+            long jvmStartTimeOffset = System.currentTimeMillis() - (System.nanoTime() / 1000000L);
```

Comment:
Since System.nanoTime is unrelated to currentTimeMillis, use exact arithmetic to avoir overflows. Explain in comment what this code is doing, it is not that clear.

Reply:
Both are positive, both are long. How can this operation lead to an overflow ?
Math.xxxExact throw exceptions, I don't want to crash on overflow, but managed it to be able to use RUNNERIN and RUNNERAT with MAXLONG

Reply:
Moved timeMillis to NanoTime conversion into TimeSource, to be consistent.